### ISSUE TITLE
[FEAT] Allow assets to be stored and loaded from an S3 store

### ISSUE THREAD

**Issue by donicrosby** [Kruptein/PlanarAlly#500]
**Is your feature request related to a problem? Please describe.**
Allowing the server to store and load assets from an S3 bucket would help server admins to not have to worry about storing large amounts of files on the web server and reduce the cost to run an instance on a cloud provider or low storage home server. This would also allow users to host the images on a self-hosted Minio instance pointing to their bulk storage without the need to setup an NFS or SAMBA mount.

**Describe the solution you'd like**
The Boto3 library is a popularly used library for working with S3 buckets, the license is compatible with the project and does a lot of the heavy lifting for the developer. 
In the configuration file you could add an option for storage backend to be either local or s3. If the backend is set to S3 have a section to set the hostname (if blank Boto assumes you're using AWS), and your Access Key and Access Secret Key.

**Describe alternatives you've considered**
Allow the user to setup a way to store their assets remotely so that they don't have to host a lot of storage on the web server.

**Additional context**
None

**Comment by Kruptein** [comment]
I can see this being useful, but I'm going to be upfront and say that this is not going to be something I'm going to pick up in the near future, because there are some other more fundamental things I wish to prioritize first.

**Comment by donicrosby** [comment]
That's fine, I may implement it when I have time and make a pull request if that's alright with you.

**Comment by Kruptein** [comment]
Absolutely, I do need to make some work of starting to document stuff though :D

**Comment by 10leej** [comment]
Can you mount the S3 store with SSHFS? You can do that with google cloud. I haven't looked into storage though since my server right now only has 10GB of data on it.
